### DISCUSSION TITLE
开发通过API创建文章的功能

### DISCUSSION THREAD

**Discussion by longredhao** [halo-dev/halo#6343] [Q&A]
### 你当前使用的版本

2.16.3

## 为什么
有时候发送固定排版格式的文章，文章段落，排版格式相同，只是里面的数据不同。若这时候每次都需要人工登陆，修改文章的数据，人工发送文章 ，效率大大下降。
参考了 官方接口文档：https://api.halo.run/  。只发现了获取文章的API，没有创建文章/发布文章/修改文章 的API
## 诉求
希望开发  创建文章，发布文章，修改文章的API。通过access-key访问，这样可以通过调用halo的API发布文章，大大提高效率

**Comment by ruibaby** [comment]
Halo 目前提供了完整的 API，建议参考：

1. https://demo.halo.run/webjars/swagger-ui/index.html
2. https://www.npmjs.com/package/@halo-dev/api-client

> 希望开发 创建文章，发布文章，修改文章的API。通过access-key访问，这样可以通过调用halo的API发布文章，大大提高效率

现在支持个人令牌，可以参考：https://docs.halo.run/user-guide/user-center#%E4%B8%AA%E4%BA%BA%E4%BB%A4%E7%89%8C

> 参考了 官方接口文档：[api.halo.run](https://api.halo.run/)

这是 Halo 1.x 的 API 文档了。

/kind support

**Comment by longredhao** [comment]
@ruibaby  谢谢。看了 https://demo.halo.run/webjars/swagger-ui/index.html 。接口数量非常多
请问下，若只是发布一篇文章，请问用哪一个接口 （或几个接口） 。有类似说明文档吗 ？

**Comment by JohnNiang** [comment]
Hi @longredhao ，推荐这个 API：<https://demo.halo.run/webjars/swagger-ui/index.html#/PostV1alpha1Uc>。

**Comment by kevin518** [comment]
@longredhao 请问部署的环境怎么才能支持openapi

**Comment by JohnNiang** [comment]
可以添加以下配置来启用 API doc：
```yaml
springdoc:
  api-docs:
    enabled: true
```
